### ISSUE TITLE
Add Monero to fiat trading pairs using BTC as the security deposit (multi-sig)

### ISSUE THREAD

**Issue by pazza83** [bisq-network/projects#50]
## Description

This is a project proposal to create XMR/Fiat markets on Bisq using BTC as the security deposit. 

I have documented this proposal following discussion of it on Busq's GitHub discussions: [#5285 ](Monero to fiat trades using BTC as the multi-sig #5285)

## Rationale

XMR/BTC is the most liquid market on Bisq. This market generates more revenue for Bisq than all of the other markets combined. Efforts are being made with atomic swaps that when/if implemented will make the future of this market, and the revenue generated from it, uncertain.

Being able to use Bisq for XMR/Fiat trades is overwhelming popular with Monero community. See my recent Reddit Post [Would you like to be able to use Bisq to buy Monero directly with fiat?](https://www.reddit.com/r/Monero/comments/micevf/would_you_like_to_be_able_to_use_bisq_to_buy/)

### Why is it important? 

The project is important for a number of reasons. These include:

There are currently no ways (at least that I know of) for people buy or sell Monero using fiat on a decentralized exchange and with no KYC requirements. Options seem to be centralized exchanges (eg Binance and Kraken) or centralized P2P exchanges such as Local Monero or AgoraDesk. These options require KYC.

On Monero community are also frequently asking about how to buy XMR using fiat and get referred to a centralized exchange such as Kraken or to a P2P marketplaces. Bisq gives great on and off ramps to BTC user, I think it could extend these on and off ramps to Monero users.

Allowing users to buy and sell in peer to peer on a decentralized exchange is core to the principles of Bisq. Enabling Monero to fiat trades on Bisq would be of a major benefit to the Monero community, It would be well used and would increase Bisq trading revenues.

- Enable people to buy Monero with fiat with no KYC on a decentralized exchange 
- Enable people to sell Monero for fiat with no KYC on a decentralized exchange 
- Enable Bisq to increase trade revenues (Monero to fiat)
- Bring more Monero users to the platform for (XMR/BTC trades, Bisq main revenue source)
- If it proves popular, it would be a good base to then develop XMR as a base pair for Bisq (see below for why I think adding more base pairs are needed for the long term success of Bisq).
- Also if the protocol can be developed to allow BTC as the trade pair it opens up the options of anything being traded for anything. I think Bisq being used as an escrow service has a lot of possibilities. I will create a separate post about this in the future. 

### Why should it be done now? 

When I started using Bisq I did so because of the great on / ramps for fiat it provides. It is fantastic for people looking to acquire some BTC in decentralized way without KYC. Bisq is still great for this use case. However the more the crypto landscape has changed (big increases in prices across the board), the more I think Fiat > BTC is not where Bisq will get most of it's revenue from in the future. I think the largest revenue and growth for Bisq can come from users trading one crypto for another. My project might be about Monero to fiat trades, but really to goal is for Bisq to strengthen its appeal and use case for another community other than Bitcoin. If this model is successful there would be no reason to use it as a way to onboard other crypto communities into Bisq. I feel like Monero is the best fit for Bisq due to it sharing Bisq's values in terms of privacy and also the high number of Monero users already on the platofrm.

I want to see Bisq succeed in growing its trading volume in BTC terms. I want it to grow it's revenue in BTC terms. If Bisq is reliant on being an on/off ramp for fiat, the increase in the price of Bitcoin is a threat to Bisq achieving this. The higher the prices the more risk there is, and the more users will run into bank imposed limits (both in terms of fiat volume and individual fiat transaction amounts). I see Bisq as needing to move into more markets to increase it's trade volume in BTC terms. The more markets it has the easier it is to generate revenue both as an on / off ramp for fiat and also the a place to go for decentralized. no KYC P2P crypto to crypto. 

The speed of price increased in the crypto space is very fast and this is why I think that efforts should be made now to onboard more communities.

Bisq is currently has a great reputation in the Monero community and they are in need of a solution to buy and sell XMR with no KYC. Bisq should see this as an opportunity and capitalize on it.

### What will happen if we don't do it or delay doing it? 

If nothing is done to address the issues above than I worry that Bisq will be unable to realistically grow it's denominated BTC revenues relying on fiat trades alone. I make the assumption that growing Bisq in these terms is something the majority pf Bisq stakeholders want but I imagine there are a range of opinions on the philosophical goals for Bisq and whilst generating revenue and being self sustaining are important, their may be differing views on how important, or not, it is to grow in BTC terms.

### Current threats of not doing anything are:

- It is only a matter of time before new exchanges are created, or existing exchanges are adapted, to cater for Monero users wanting to buy / sell using fiat with no KYC. This could threaten Bisq's XMR/BTC markets and revenues. It could also threaten some of the current BTC/fiat sellers. For example if the mempool is too high traders could instead trade XMR>fiat for lower fees. 
- XMR/BTC atomic swaps when implemented would likely threaten Bisq's XMR/BTC markets and revenues.
- BTC prices are likely continue to rise resulting in a decrease in BTC denominated revenue for Bisq the DAO. This will affect Bisq's growth.

## Criteria for delivery

My project proposal is to create XMR/fiat trade where BTC is used as the multi-sig. 

I have chosen to keep BTC as the multi-sig for the following reasons:

- It reduces development time.
- It can be a template for adding other altcoins
- It is a proof of concept for trading anything for anything using BTC as a multi-sig
- Has the benefit of all the security that BTC brings with it.

### What would a XMR/Fiat trade look like?

Sellers of XMR would deposit 115% BTC to cover trade amount and security deposit. When trade complete seller would receive their full security deposit back.

Buyers of XMR would deposit 15% BTC to cover security deposit. When trade complete buyer would receive their full security deposit back.

#### Example of user Selling XMR

Example for 10 XMR trade (€1,900)

- Sell XMR / Buy Fiat
- XMR seller makes an offer to sell 10 XMR for €1,900
- XMR seller would deposit 115% of trade amount in BTC (0.046 BTC)
- When XMR buyer accepts offer they would deposit 15% of trade amount in BTC (0.006 BTC)
- 0.052 BTC locked in multi-sig
- XMR buyer sends €1,900 to XMR Seller's bank account
- XMR seller confirms receipt of €1,900, sends 10 XMR
- XMR buyer confirms 10 XMR have been received (or trade auto-confirmed)
- BTC deposit is released to buyer and seller
- If trade goes wrong it can enter mediation

#### Example of user Buying XMR

Example for 10 XMR trade (€1,900)

- Buy XMR / Sell Fiat
- XMR buyer makes an offer to buy 10 XMR for €1,900
- XMR buyer would deposit 15% of trade amount in BTC (0.006 BTC)
- When XMR seller accepts offer they would deposit 115% of trade amount in BTC (0.046 BTC)
- 0.052 BTC locked in multi-sig
- XMR buyer sends €1,900 to XMR Seller's bank account
- XMR seller confirms receipt of €1,900, sends 10 XMR
- XMR buyer confirms 10 XMR have been received (or trade auto-confirmed)
- BTC deposit is released to buyer and seller
- If trade goes wrong it can enter mediation

### Checklist

- [ ] Get feedback from Bisq / Monero community regarding project
- [ ] Confirm trade protocol can be changed to enable to above 
- [ ] Confirm what changes to GUI would be needed (more market pairs etc)
- [ ] Confirm a developer that has the resources to complete the project
- [ ] Develop any necessary user experience designs for the dev to work with
- [ ] Confirm any in app text changes / additions needed
- [ ] Developer to complete changes to protocol
- [ ] Test 
- [ ] Deploy
- [ ] Write and publish wiki on XMR/Fiat  
- [ ] Promote XMR/Fiat market to Bisq / Monero community
- [ ] List trade pair with services like Coin Geko / Coin Market Cap etc
- [ ] Troubleshoot
- [ ] Retrospective analysis

At the end of the project Bisq users will be able to buy and sell Monero for fiat with no KYC on a decentralized exchange.

## Measures of success

Success would be measured in terms of trade volume on the following markets:

- XMR/Fiat (no trade revenue exists so all revenue would be additional)
- XMR/BTC volume (increase would be expected overtime as more Monero users use Bisq)

## Risks

There would likely be a risk to adding a new trade protocol for the 115% deposit. There would also likely be some work required to maintain this. I would defer to the devs for assessment of these risks.

Monero also likely brings with it more regulatory risks. I think I would be right in saying more countries have banned the trading of Monero than they have the trading of BTC. For example I think exchanges based in Australia had pressure placed upon them to remove the ability for users to purchase Monero.  My view point, however, is the regulatory risks are not a reason for this to be avoided, in fact I think they are a reason to push forward. It reminds me of the poem [First_they_came_... ](https://www.youtube.com/watch?v=4sEXWjTyAbA)

## Estimates

I would be happy to do this alongside a dev. But if someone else wanted to project manage it I would be happy to defer to them. If the idea is approved in principle I would be happy to agree a fixed USD amount for completing all the aspects.

I am not sure what the dev budget would be. 

## Notes

This project is not for XMR to be added as base currency, but it is the logical next step should this project be successful. I have not been contributing long enough to have been involved in the previous [Propose how to integrate Monero wallet into Bisq](https://github.com/bisq-network/bisq/issues/1863), [Bisq Monero Wallet](https://github.com/bisq-network/proposals/issues/110) and the rejection of the proposal [Bisq DAO Cycle 6: Results](https://bisq.network/blog/cycle-6-results/).

Therefore, I might be missing some of the history/information, and I don't know if my proposal is significantly different, or if things have changed. 

I am making this proposal as I think that revenue from fiat to BTC on/off ramps revenues will likely decrease over the coming years and I for Bisq to succeed in the long term I see it as a place where more crypto to crypto exchanges can take place. I am also passionate about privacy, decentralization, open societies, and the ability to Bitcoin and other cryptos to help people achieve personal and financial freedom. I feel is this project where to implemented it would be of benefit to both the Bisq and Monero community.

**Comment by wiz** [comment]
I don't think this is feasible without forking the trade protocol. It would be quite a large undertaking.

**Comment by pazza83** [comment]
Would there be a work around, increase the percentage that could be used for the security deposit to 115%?

**Comment by m52go** [comment]
I think this is an interesting idea but potentially flawed.

### XMR-BTC

I realize this proposal is not about XMR-BTC, but this background might be helpful for perspective. 

Bisq is already capable of doing high XMR trading volumes. Back in summer 2019 it wasn't uncommon to see 80, 100, or even 200 BTC trading days. Nowadays it's rare to see a 50 BTC trading day, but it does happen. It's just not clear what drives these spikes. 

I spent time last year trying to determine reasons, but best I could find was an XMR Telegram group where a Bisq enthusiast would post about Bisq every Monday, offer help, etc. Not sure this is actually what drove the crazy volume in those days, but the time periods of this activity align with the spikes in trading volume.

Another thought (and one I think makes more sense) is that the "new" trade protocol (the one in use now, introduced in v1.2 with mediation and delayed payout transactions) killed the feasibility of active XMR trading on Bisq. If you look back at XMR volume spikes, you'll notice that they tend to be for 1 or 2 BTC trades, and [many of them were instant](https://github.com/bisq-network/roles/issues/38#issuecomment-599561611) (1-hour time window) trades. This indicates that traders wanted to trade big amounts quickly. If there were issues, arbitrators (old trade protocol) would typically solve within 24 hours, and traders depended on that. With the current trade protocol, an altcoin dispute which cannot be solved by mediation necessarily takes 10+ days to resolve. And that's just to engage an arbitrator -- actually getting funds back takes much longer -- and even after all that, funds are in BSQ, not BTC or XMR. 

Imagine the possibility of having 1 to 2 BTC of trading capital tied up for several days (maybe weeks) because of a stupid trading mistake the other guy made...

### XMR-fiat

Feedback I've seen and heard from XMR folks in the past indicates that any exchange mechanism touching BTC is a problem...so frankly I'm surprised at the positive response you got for this idea from the XMR community. I get the impression a lot of XMR people like the idea of Bisq but have been turned away because they perceive it to be anti-XMR, so any pro-XMR news is cheered no matter what (e.g., the "the Mom test"). For example, I remember the XMR autoconfim feature getting lots of positive reactions but ultimately it doesn't seem like it's used much.

Anyway that's just speculation on my part. But still, this idea doesn't seem to solve the core problem XMR people tend to have with Bisq. If it's not much effort to implement, I wouldn't oppose giving it a shot, but if it does take a lot of effort, I would be in favor of validating the idea more rigorously somehow first.

**Comment by pazza83** [comment]
Hi @m52go thanks for your comments.

**XMR-BTC**

Thanks for the background with regards XMR-BTC. I agree that the arbitration time and refunds given in BSQ would deter many users from trading on Bisq, especially if those trades where over the amount the refund agent can compensate.

I wonder if the large trades come at times when there are good arbitration opportunities between Bisq and other exchanges. That is why trades need to be instant to recycle the trade amount quickly to take advantage of the market conditions.  The nature of trading on Bisq means there is no slippage, that and the XMR/BTC price updating only every ~60 seconds?  would make Bisq a good platform for arbitrage when conditions are right on other exchanges.  

Being the current main source of trading fee revenue for the DAO, it would be great if Bisq could grow this market. There are a number of competing centralized, decentralized, and variations in-between, alternatives for XMR/BTC trading that does not require KYC. 

Since I posted on Reddit there has been another thread about a Bisq fork [Monero based DEX for trading XMR for fiat and cryptocurrencies! (PoC)](https://www.reddit.com/r/Monero/comments/mmrchk/im_happy_to_announce_haveno_the_monero_based_dex/). 

There is also the issue of atomic swaps happening in the future. 

What can be done to make Bisq the go to place for XMR/BTC trading? 

**XMR-fiat**

Yes feedback was overwhelmingly positive. There were a couple of comments from people saying they would not use it because of the BTC involved, but the vast majority where for it. No exchange will ever please everyone. 

Currently there exists a gap for people wanting to buy / sell XMR for fiat. There seems to be a market demand for this gap to be filled. What is to stop Bisq being the platform to fill this gap?

A number of comments on the Reddit post referenced they would be happy to donate for Bisq to integrate an XMR wallet / trade protocol.  Others mentioned about Bisq putting a bounty / proposal together to incentivize a developer familiar with XMR multi-sigs to complete the work. I think this would be a good use of BSQ issuance and would likely be repaid in new revenue for the DAO.  

> I remember the XMR autoconfirm feature getting lots of positive reactions but ultimately it doesn't seem like it's used much.

At the time of me posting this all the take offer to sell XMR are using the autoconfirm. No one currently making an offer has auto confirm disabled.

**Comment by m52go** [comment]
Ultimately I think it boils down to what Bisq stakeholders want Bisq to be. If they want it to be an effective XMR exchange, they should spend resources to take the project in that direction. If they want it to be an effective fiat exchange, they should spend resources to take the project in that direction.

I'm not sure how practical it is for Bisq to cater to both...would need a developer to clarify how much in the way of resources the project proposed in this thread would require. XMR atomic swaps could help, as a marketplace would still be needed to match buyers and sellers, so Bisq could serve that function when the technology is available (maybe it could even serve as a liquidity pool for wallets and other tools that offer swaps).

Of course altcoin trading is profitable...but is investing in it the preferred way ahead for the project? If given 100,000 USD and 6 months of full-time work of a brilliant developer, I would imagine most stakeholders would opt to spend those resources on moving the Bisq trade protocol off of layer one, building out the API, improving usability, overhauling the payment method implementation, etc instead of building out XMR trading functionality.

**Comment by pazza83** [comment]
Yes understand that it needs to grow in the way the stakeholders want it too.

I think Bisq as a fiat > BTC exchange is core to what Bisq is. 

I think if Bisq wants to grow in terms of BTC revenue it will need to develop markets for other altcoins. XMR seems like a great fit for Bisq and think there is the opportunity there for Bisq to cater for the XMR communities needs more than it does at present.

> Of course altcoin trading is profitable...but is investing in it the preferred way ahead for the project? If given 100,000 USD and 6 months of full-time work of a brilliant developer, I would imagine most stakeholders would opt to spend those resources on moving the Bisq trade protocol off of layer one, building out the API, improving usability, overhauling the payment method implementation, etc instead of building out XMR trading functionality.

Agreed, but I do not see a reason why both can not be achieved. It might require initial 'investment' through BSQ issuance but I think this would be paid back though an increase in revenue.

**Comment by m52go** [comment]
> Agreed, but I do not see a reason why both can not be achieved. It might require initial 'investment' through BSQ issuance but I think this would be paid back though an increase in revenue.

Shortage of human capital, not financial capital. I don't see a way to do this without diverting developer resources from other top priorities. Even if an outside developer is found to do the work, Bisq maintainers would still need to spend time to review that work...which could be significant if it involves major protocol changes, which is why this idea really needs developer feedback. Maybe the required changes are minor and I'm blowing this out of proportion.

**Comment by chimp1984** [comment]
I think that would be a potential protocol candidate for Bisq 2. Once we have the Bisq Multisig implemented we could make a variation our of it without the sellers BTC amount and have the XMR-Fiat protocol. That should not be too much effort.
For Bisq 1 I think the effort is too high.

**Comment by HardenedSteel** [comment]
Now the required effort shouldn't be too high because of Haveno fork. However BTC involvement makes this proposal useless.

I would like to hear what's the status of this proposal?

**Comment by pazza83** [comment]
Closing as not completed / stale
